<?xml version="1.0" encoding="UTF-8"?>
<launch>

   

  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find projet)/worlds/Projet2024.world"/>
    <arg name="paused" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="true"/>
    <arg name="headless" value="false"/>
    <arg name="debug" value="false"/>
  </include>


  <!-- Spwan turtlebot in Gazebo -->

  <param name="robot_description" command="$(find xacro)/xacro $(find projet)/urdf/turtlebot3.urdf.xacro " />

  <node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf" args="-urdf -model turtlebot3_burger -x 0.986 -y 1.69 -z 0.0 -Y -1.618 -param robot_description" />

  <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher"/>
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"/>
  
  <node name="line_follower" pkg="projet" type="EnchainnementPresqueParfait.py" output="screen" required="true"/>
      <param name = "speed_linear" value= "0.1"/>
      <param name = "speed_angular" value= "0.5"/>
      <param name = "angle_detection" value= "50"/>
      <param name = "distance_limit" value= "0.325"/>
      <param name = "test" value="test" type="str"/>
      <param name = "distance_critique" value="0.2"/>
      <param name = "emergency_stop" value = "False" type ="bool"/> 
      <param name = "red_lines_counter" value = "2"/> 
      <param name = "N_launch_file" value = "2"/> 
      


</launch>
